Add hook for the bucket settings navgiation

ekala opened the pull request:
- The hook allows plugins to add navigations links on the settings
  navigation toolbar by introducing a new event: `swiftriver.bucket.settings.nav` Staging for #26
- We pass off an array with the `$bucket_base_url` and `$active` as the event data. This allows the plugins that hook into this event to access the URL of the bucket and also enables them manipulate the value of `$active`

ekala committed:
Add hook for the bucket settings navgiation

* The hook allows plugins to add navigations links on the settings
navigation toolbar. Staging for #26

69mb commented on themes/default/views/pages/user/layout.php:
```diff
@@ -34,7 +34,7 @@
 </nav>
 <?php endif; ?>
 
-<div id="content" class="user <?php echo $view_type ;?> cf">
```
Why are you removing the user class?

ekala replied:
The user class distorts the alignment of the action buttons in the deployments listing (introduced by the Ushahidi push plugin) on the user dashboard i.e.

![user Class Present](https://dl.dropbox.com/u/2635815/user-class-present.png)

**With the class removed:**

![user class removed](https://dl.dropbox.com/u/2635815/user-class-removed.png)

The visual impact on the other sections of the dashboard is a slight increase in the height of the settings containers

69mb replied:
This style is lost and the container titles have too much padding around them:

``` css
#content.user article.container div.property-title {
  padding: 5px 0;
}
```

Perhaps improve this one or add one to correct the alignment?

``` css
#content.user article.container a.remove-large {
  margin: 13px 10px 13px 0;
}
```

ekala committed:
Restore "user" CSS class

* Added an extra CSS selector that is specific to the <div> actions
container(s) that appear in the <header> section of the container titles

* Added two new event hooks to be run when a drop is added or removed
from a bucket.

ekala commented:
@69mb Made the css changes to:
- Preserve the padding around the container titles
- Accomodate having  action buttons in the container title section

69mb commented:
+1
